src/averageAges.js

```diff
@@ -14,12 +14,30 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
+const SEX_MALE = 'm';
+const SEX_FEMALE = 'f';
+
+function calculateMenAverageAge(people, century = null) {
   // write code here
   // learn how to use array methods like .filter .map .some .every .find .reduce
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const centuryValue = 100;
+  const men = people
+    .filter(person => person.sex === SEX_MALE);
+  const filteredMen = century !== null
+    ? men
+      .filter(person => Math.ceil(person.died / centuryValue) === century)
+    : men;
+
+  const amountAge = filteredMen
+    .reduce((sum, person) =>
+      sum + (person.died - person.born), 0);
+
+  const avarageMenAge = amountAge / filteredMen.length;
```

Comment:
Similar calculations are made for women, so it should be transferred to a separate function and reused.